internal/core/window.rs

```diff
@@ -431,8 +434,8 @@ pub struct WindowInner {
     maximized: Cell<bool>,
     minimized: Cell<bool>,
 
-    active_popup: RefCell<Option<PopupWindow>>,
-    had_popup_on_press: Cell<bool>,
+    active_popups: RefCell<BTreeMap<u32, PopupWindow>>,
```

Comment:
I think you can keep that a Vec (from the my patch), and store the id in the `struct PopupWindow`. It is fine to do a linear search for the seldom case where we need to look it up by id in close. (we don't expect many popup visible at the same time anyway)

Reply:
Will do!